src/components/ui/EmailForm.tsx

```diff
@@ -1,50 +1,79 @@
-import { width } from 'classnames/tailwind'
+import { AccentText } from 'components/ui/Text'
+import { textAlign } from 'classnames/tailwind'
+import { useSnapshot } from 'valtio'
 import Button from 'components/ui/Button'
 import Email from 'icons/Email'
+import EmailFormStore from 'stores/EmailFormStore'
 import GradientBorder from 'components/ui/GradientBorder'
 import Input from 'components/ui/Input'
-import useEmailForm from 'hooks/useEmailForm'
+import LoadedFilesList from 'components/ui/LoadedFilesList'
 
 export default function ({
   loading,
   onSubmit,
   submitType = 'primary',
   placeholder = 'Enter...',
-  submitText = 'Submit',
 }: {
   loading?: boolean
   submitType?: 'primary' | 'secondary' | 'tertiary'
-  onSubmit: (email: string) => void
-  submitText?: string
+  onSubmit: (emails: readonly string[]) => void
```

Comment:
`readonly` looks weird. I guess if you pass `[...emails]`, it will solve the issue:
```
onSubmit([...rawEmails])
```